src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt

```diff
@@ -124,4 +125,94 @@ class SmartDocumentsTemplatesService @Inject constructor(
                 )
         ).resultList.toSet().toRestSmartDocumentsTemplateGroup()
     }
+
+    /**
+     * Get the information object type UUID for a pair of group-template in a zaakafhandelparameters
+     *
+     * @param zaakafhandelParametersUUID UUID of a zaakafhandelparameters
+     * @param templateGroupName name of a template group
+     * @param templateName name of a template under the group
+     * @return information object type UUID associated with this pair
+     */
+    fun getInformationObjectTypeUUID(
+        zaakafhandelParametersUUID: UUID,
+        templateGroupId: String,
+        templateId: String
+    ): UUID {
+        LOG.info {
+            "Fetching information object type UUID mapping for zaakafhandelParameters UUID " +
+                "$zaakafhandelParametersUUID, template group id $templateGroupId and template id $templateId"
+        }
+
+        val zaakafhandelParametersId =
+            zaakafhandelParameterService.readZaakafhandelParameters(zaakafhandelParametersUUID).id
+        val builder = entityManager.criteriaBuilder
+        val query = builder.createQuery(SmartDocumentsTemplate::class.java)
+        val root = query.from(SmartDocumentsTemplate::class.java)
```

Comment:
Lately I tend to use `.let { .. }` everywhere instead of declaring specific `vals` but I am in doubt what is best practice in a case like this.. Maybe we should discuss this some time. :-)